src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala

```diff
@@ -98,16 +100,69 @@ class HouseholdFleetManager(
-        case firstVehicle :: rest =>
-          logger.debug("Vehicle {} is now taken", firstVehicle.id)
-          firstVehicle.becomeDriver(sender)
-          sender() ! MobilityStatusResponse(Vector(ActualVehicle(firstVehicle)), triggerId)
-          rest
-        case Nil =>
-          sender() ! MobilityStatusResponse(Vector(), triggerId)
-          Nil
+    case MobilityStatusInquiry(personId, whenWhere, _, requireVehicleCategoryAvailable, triggerId) =>
+      {
+        for {
+          neededVehicleCategory              <- requireVehicleCategoryAvailable
+          emergencyHouseholdVehicleGenerator <- maybeEmergencyHouseholdVehicleGenerator
+          vehicle                            <- emergencyHouseholdVehicleGenerator.createVehicle(personId, nextVehicleIndex, neededVehicleCategory)
+        } yield {
+          val vehicleCreatedOutOfThinAir: Boolean = if (availableVehicles.isEmpty) {
+            logger.warn(
+              s"No vehicles available for category ${neededVehicleCategory} available for person ${personId.toString}, creating a new vehicle with id ${vehicle.id.toString}"
+            )
+            emergencyHouseholdVehicleGenerator.createVehicle(personId, nextVehicleIndex, neededVehicleCategory) match {
+              case Some(vehicle) =>
+                nextVehicleIndex += 1
+                vehicle.setManager(Some(self))
+                vehicle.spaceTime = whenWhere
+                val mobilityRequester = sender()
+                vehiclesInternal(vehicle.id) = vehicle
+
+                // Pipe my car through the parking manager
+                // and complete initialization only when I got them all.
+                val responseFuture = parkingManager ? ParkingInquiry.init(
+                  whenWhere,
+                  "wherever",
+                  triggerId = triggerId
+                )
+                responseFuture.collect { case ParkingInquiryResponse(stall, _, otherTriggerId) =>
+                  vehicle.useParkingStall(stall)
+                  logger.debug("Vehicle {} is now taken, which was just created", vehicle.id)
+                  vehicle.becomeDriver(mobilityRequester)
+                  MobilityStatusResponse(Vector(ActualVehicle(vehicle)), otherTriggerId)
+                } pipeTo mobilityRequester
+                true
+              case _ =>
+                false
+            }
+          } else false
+
+          if (!vehicleCreatedOutOfThinAir) {
+            availableVehicles = availableVehicles match {
+              case firstVehicle :: rest =>
+                logger.debug("Vehicle {} is now taken", firstVehicle.id)
+                firstVehicle.becomeDriver(sender)
+                sender() ! MobilityStatusResponse(Vector(ActualVehicle(firstVehicle)), triggerId)
+                rest
+              case _ =>
+                logger.error(s"THE LIST OF VEHICLES SHOULDN'T BE EMPTY")
+                Nil
+            }
+          }
+        }
+      }.getOrElse {
```

Comment:
I am just now noticing that this is an `Option[Unit]`, which means it is basically working off of the fact whether the `for` did something at all. Is that what is wanted? If so, then maybe it is better to assign to a variable and then do an if isEmpty. Ultimately it is more about confusing a reader of the code, so it will be functionally the same and not worth blocking., though